src/resolutions/lesson-02/2-type-test.ts

```diff
@@ -0,0 +1,22 @@
+import { Equal, Expect } from "../../../helper";
+/*2. 
+    same as above but now we want a default type string if nothing is passed,
+	notice the test is also asking us to type it as Set<string> when no generic is passed 
+    to the function. 
+*/
+
+type SetType = string | number;
+
+export const createOwnSet = <T extends SetType = string>(t: Iterable<T> = []) => {
+  return new Set(t);
+};
+
+const stringSet = createOwnSet<string>(["set"]);
+const numberSet = createOwnSet<number>([1]);
+const defaultString = createOwnSet();
+
```

Comment:
1. no es necesario asignar a nuestro argumento el tipo `Iterable<T> = []`, ya que solo necesitamos pasarle un string o numero, un iterable es distinto. ademas por la sintaxys que es igual a un arrat [] confunde a los que solo han trabajado con JS.  
ejemplo: esto `["set"]` es un array de string, no un string para JS.

Reply:
Thx, addressed in next commit.